tests/phpunit/integration/Core/Authentication/Google_ProxyTest.php

```diff
@@ -82,23 +88,72 @@ public function test_get_user_fields_for_network_administrator() {
 		);
 	}
 
-	public function test_sync_site_fields() {
+	public function test_unregister_site() {
 		$context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
 		$google_proxy = new Google_Proxy( $context );
 		$credentials  = new Credentials( new Options( $context ) );
 
-		add_filter( // Fake "using proxy"
-			'googlesitekit_oauth_secret',
-			function () {
-				return array(
-					'web' => array(
-						'client_id'     => '12345678.apps.sitekit.withgoogle.com',
-						'client_secret' => 'test-client-secret',
-					),
-				);
-			}
+		$fake_creds = $this->fake_proxy_site_connection();
+
+		$pre_args = null;
+		$pre_url  = null;
+
+		add_filter(
+			'pre_http_request',
+			function ( $false, $args, $url ) use ( &$pre_args, &$pre_url ) {
+				$pre_args = $args;
+				$pre_url  = $url;
+
+				return $false;
+			},
+			10,
+			3
+		);
+
+		$expected_success_response = array( 'success' => true );
+		$this->expect_request_response(
+			$google_proxy->url( Google_Proxy::OAUTH2_DELETE_SITE_URI ),
+			$expected_success_response
+		);
+		$response_data = $google_proxy->unregister_site( $credentials );
+
+		// Ensure the request was made with the proper URL and body parameters.
+		$this->assertEquals( $google_proxy->url( Google_Proxy::OAUTH2_DELETE_SITE_URI ), $pre_url );
+		$this->assertEquals( 'POST', $pre_args['method'] );
+		$this->assertEqualSetsWithIndex(
+			array(
+				'site_id'     => $fake_creds['client_id'],
+				'site_secret' => $fake_creds['client_secret'],
+			),
+			$pre_args['body']
 		);
 
+		// Ensure success response data is correct.
+		$this->assertEquals( $expected_success_response, $response_data );
+
+		$expected_error_response = array( 'error' => "invalid 'site_id' or 'site_secret'" );
+		$this->expect_request_response(
+			$google_proxy->url( Google_Proxy::OAUTH2_DELETE_SITE_URI ),
+			$expected_error_response,
+			400
+		);
+
+		// Ensure exception with correct message is thrown for error response.
+		try {
+			$google_proxy->unregister_site( $credentials );
+			$this->assertEquals( $expected_error_response['error'], null );
```

Comment:
Better to fail explicitly here if this line isn't expected to be reached.
```suggestion
			$this->fail( 'Expected an exception to be thrown when unregistering the site' );
```